### PR TITLE
[risky]reduce ram usage of loading moe without monkey patch and fix qwen3-flash-next group_size issue

### PR DESCRIPTION
**Pull request by wenhuach21**
## Description

Please briefly describe your main changes, the motivation.

## Type of Change

Bug fix

## Related Issues

Fixes or relates to #

## Checklist Before Submitting

- [x] My code has been tested locally.
- [ ] Documentation has been updated as needed.
- [ ] New or updated tests are included where applicable.
- [ ] The CUDA CI has passed. You can trigger it by commenting `/azp run Unit-Test-CUDA-AutoRound`.

### CODE REVIEW

**Comment by wenhuach21** [comment]
**main branch, comment the monkey patch**

(autoround) wenhuach@mlp-dgx-01:~/auto-round-main$ CUDA_VISIBLE_DEVICES=2 python3 -m auto_round /models/Qwen3.6-35B-A3B --iters 1  --output_dir "/data2/wenhuach" --tasks lambada_openai --nsamples 8 --seqlen 256
2026-09-03 16:33:40 INFO main.py L309: start to quantize /models/Qwen3.6-35B-A3B
Loading weights: 100%|██████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 1026/1026 [00:00<00:00, 6313.53it/s]
2026-09-03 16:33:45 WARNING logging.py L340: reset enable_torch_compile to `False` as `iters`=1 is below 10, so compilation cost would outweigh its benefit
2026-09-03 16:33:45 INFO base.py L2127: format is not set, using default auto_round format.
2026-09-03 16:33:45 WARNING logging.py L340: some layers are skipped quantization (shape not divisible by 32): model.language_model.layers.[0-39].mlp.shared_expert_gate, model.visual.blocks.[0-26].mlp.linear_fc1, model.visual.blocks.[0-26].mlp.linear_fc2
2026-09-03 16:33:45 INFO replace_modules.py L128: Experts (before replacement) [model.language_model.layers.0.mlp.experts] (Qwen3_5MoeExperts):
Qwen3_5MoeExperts(
  (act_fn): SiLUActivation()
)
[transformers] `loss_type=None` was set in the config but it is unrecognized. Using the default loss: `ForCausalLMLoss`.
**2026-09-03 16:33:46 INFO device.py L1446: Before applying custom replacements 'peak_ram': 1.39GB
2026-09-03 16:33:54 INFO moe_experts_interface.py L681: [MoE Prep] Unfused 40 MOE experts modules
2026-09-03 16:33:55 INFO device.py L1446: After applying custom replacements 'peak_ram': 81.59GB**
2026-09-03 16:33:55 INFO replace_modules.py L101: Prepared 40 MOE modules for quantization
2026-09-03 16:33:55 INFO replace_modules.py L128: Experts (after replacement/skip) [model.language_model.layers.0.mlp.experts] (Qwen3_5MoeExperts):
Qwen3_5MoeExperts(
  (act_fn): SiLUActivation()
  (0-255): 256 x _ExpertContainer(
    (down_proj): Linear(in_features=512, out_features=2048, bias=False)
    (gate_proj): Linear(in_features=2048, out_features=512, bias=False)
    (up_proj): Linear(in_features=2048, out_features=512, bias=False)
  )
)
2026-09-03 16:33:55 INFO base.py L988: Using predefined ignore_layers: model.language_model.layers.[0-39].mlp.gate
2026-09-03 16:33:55 INFO layer_config_resolver.py L401: Ignored layers: model.language_model.layers.[0-39].mlp.gate
2026-09-03 16:33:56 INFO layer_config_resolver.py L401: Ignored layers: model.language_model.layers.[0-39].mlp.gate


**this branch**
(autoround) wenhuach@mlp-dgx-01:~/auto-round$ CUDA_VISIBLE_DEVICES=2 python3 -m auto_round /models/Qwen3.6-35B-A3B --iters 1  --output_dir "/data2/wenhuach" --tasks lambada_openai --nsamples 8 --seqlen 256
2026-09-03 16:35:04 INFO main.py L309: start to quantize /models/Qwen3.6-35B-A3B
2026-09-03 16:35:04 INFO model.py L348: Fused-MoE checkpoint detected: building a meta skeleton and materializing weights per block (set AR_DISABLE_AUTO_META_LOAD=1 to load the whole model on CPU instead).
**2026-09-03 16:35:07 INFO device.py L1446: Before applying custom replacements 'peak_ram': 1.1GB
2026-09-03 16:35:09 INFO moe_experts_interface.py L745: [MoE Prep] Unfused 40 MOE experts modules
2026-09-03 16:35:10 INFO device.py L1446: After applying custom replacements 'peak_ram': 1.3GB**
2026-09-03 16:35:10 INFO replace_modules.py L102: Prepared 40 MOE modules for quantization
2026-09-03 16:35:10 INFO replace_modules.py L161: Experts (before unfuse) [model.language_model.layers.0.mlp.experts] (Qwen3_5MoeExperts):
Qwen3_5MoeExperts(
  (act_fn): SiLUActivation()
)
2026-09-03 16:35:10 INFO replace_modules.py L162: Experts (after unfuse) [model.language_model.layers.0.mlp.experts] (Qwen3_5MoeExperts):
Qwen3_5MoeExperts(
  (act_fn): SiLUActivation()
  (0-255): 256 x _ExpertContainer(
    (down_proj): Linear(in_features=512, out_features=2048, bias=False)
    (gate_proj): Linear(in_features=2048, out_features=512, bias=False)
    (up_proj): Linear(in_features=2048, out_features=512, bias=False)
  )
)
2026-09-03 16:35:12 WARNING logging.py L340: reset enable_torch_compile to `False` as `iters`=1 is below 10, so compilation cost would outweigh its benefit
2026-09-03 16:35:12 INFO base.py L2066: format is not set, using default auto_round format.
2026-09-03 16:35:13 WARNING logging.py L340: some layers are skipped quantization (shape not divisible by 32): model.language_model.layers.[0-39].mlp.shared_expert_gate, model.visual.blocks.[0-26].mlp.linear_fc1, model.visual.blocks.[0-26].mlp.linear_fc2
2026-09-03 16:35:13 INFO base.py L955: Using predefined ignore_layers: model.language_model.layers.[0-39].mlp.gate

**Comment by aquilarubra** [comment]
BUILTIN_MODULES in auto_round/modeling/fused_moe/replace_modules.py got fully commented out (llama4/deepseek_v2/qwen3_5_moe/qwen3_5_moe_text/step3p5/qwen3_omni_moe), leaving `BUILTIN_MODULES = {}`. Looks like a leftover from the "disable all moe replacements" debug commit. That breaks `_should_use_meta_skeleton`'s check for families with dedicated replacements, and `is_custom_model()`/`apply_replacements` too — these models would now get routed through the generic meta-skeleton path instead of their custom handlers. It also fails your own new test (`test_families_with_a_dedicated_replacement_are_left_alone` asserts the dict has those 4 keys). Can you uncomment that block before merging?

**Comment by wenhuach21** [comment]
> BUILTIN_MODULES in auto_round/modeling/fused_moe/replace_modules.py got fully commented out (llama4/deepseek_v2/qwen3_5_moe/qwen3_5_moe_text/step3p5/qwen3_omni_moe), leaving `BUILTIN_MODULES = {}`. Looks like a leftover from the "disable all moe replacements" debug commit. That breaks `_should_use_meta_skeleton`'s check for families with dedicated replacements, and `is_custom_model()`/`apply_replacements` too — these models would now get routed through the generic meta-skeleton path instead of their custom handlers. It also fails your own new test (`test_families_with_a_dedicated_replacement_are_left_alone` asserts the dict has those 4 keys). Can you uncomment that block before merging?

Yes, this is mainly to trigger our internal tests for more MoE models. I’d keep the monkey patch for one or two release versions and remove it eventually.

File: auto_round/modeling/fused_moe/replace_modules.py [hunk 1]
```diff
@@ -31,17 +32,17 @@
...
-    "qwen3_5_moe": LazyImport("auto_round.modeling.fused_moe.qwen3_5_moe"),
-    "qwen3_5_moe_text": LazyImport("auto_round.modeling.fused_moe.qwen3_5_moe"),
-    # Step 3.5 MoE: splits fused MoELinear into per-expert nn.Linear
-    "step3p5": LazyImport("auto_round.modeling.fused_moe.step3_5_moe"),
-    # Qwen3-Omni MoE: thinker (no shared expert)
-    "qwen3_omni_moe": LazyImport("auto_round.modeling.fused_moe.qwen3_omni"),
+    # # Llama4 has no use_experts_implementation, needs custom replacement to handle fused MoE blocks.
+    # "llama4": LazyImport("auto_round.modeling.fused_moe.llama4"),
+    # # DeepseekV2Attention enables q_scale calibration for deepseek v2 on Gaudi (#1299)
+    # "deepseek_v2": LazyImport("auto_round.modeling.fused_moe.deepseek_v2"),
```
Comment (wenhuach21) [inline auto_round/modeling/fused_moe/replace_modules.py]: @copilot  uncomment it

**Comment by wenhuach21** [comment]
Qwen3.8-flash,glm-5.3-BF16 smoke test passed

**Review by luoyu-intel — APPROVED** [review]
